### DISCUSSION TITLE
Prevent saving component when getHtml is called.

### DISCUSSION THREAD

**Discussion by banderovskyi** [GrapesJS/grapesjs#5648] [Q&A] [answered]
Hello!
I'm trying to figure out how to add a component to the 'canvas' in such a way that when 'getHtml' is called, this particular component doesn't show up. In my scenario, I need to include some inline JS code. When I add it using 'setComponents,' I find myself having to replace my components 'manually' in the resulting HTML string. Is there a way to stop grapesjs from parsing it?

Additionally, I was considering whether it's possible to insert my inline JS just before closing the body tag. Can this be done?

**Comment by artf** [comment, marked as answer]
You need to extend the toHTML method
```js
editor.Components.addType('you-custom-component', {
  model: {
    // ...
    toHTML(opts) {
      const someProp = this.get('someComponentProp');
      const domAttributes = this.getAttrToHTML();
      const innerHtml = this.getInnerHTML(opts);
      // ..
      return `<div>My custom HTML</div>`
    }
  },
});
```

> Additionally, I was considering whether it's possible to insert my inline JS just before closing the body tag. Can this be done?

Same thing as above, you need to extend the `wrapper` component.

**Reply by banderovskyi** [reply]
Thanks, it helped and solved my issue!

About extending the wrapper, I didn't get it completely how I can add scripts before closing body.  However, it wasn't my main issue.

**Reply by artf** [reply]
```js
// 'wrapper' is the component used for the body
editor.Components.addType('wrapper', {
  model: {
    // ...
    toHTML(opts) {
      const innerHtml = this.getInnerHTML(opts);
      // ..
      return `<body>
         ${innerHtml}
         <div>your custom static content</div>
      </body>`
    }
  },
});
```
